warning in quick_large_files.c

Seeing this warning. I will fix.

```
gcc -DHAVE_CONFIG_H -I. -I../..  -I../../include -I../../liblib -I/usr/local/hdf5-1.10.5/include  -Wall -g -MT filter_example.o -MD -MP -MF $depbase.Tpo -c -o filter_example.o filter_example.c &&\
mv -f $depbase.Tpo $depbase.Po
quick_large_files.c: In function ‘main’:
quick_large_files.c:44:9: warning: unused variable ‘i’ [-Wunused-variable]
     int i, res;
         ^
quick_large_files.c:42:12: warning: unused variable ‘val_in’ [-Wunused-variable]
     double val_in, val_out = 999.99;
            ^~~~~~

```

--- a/examples/C/Makefile.am
+++ b/examples/C/Makefile.am
@@ -17,7 +17,7 @@ AM_LDFLAGS = ${top_builddir}/liblib/libnetcdf.la
 # These are the netCDF-3 examples.
 check_PROGRAMS = simple_xy_wr simple_xy_rd sfc_pres_temp_wr	\
 sfc_pres_temp_rd pres_temp_4D_wr pres_temp_4D_rd format		\
-quick_large_files quick_small_files sfc_pres_temp_more
+quick_large_files quick_small_files sfc_pres_temp_more simple
 
 TESTS = run_examples.sh format quick_large_files quick_small_files	\
 sfc_pres_temp_more

--- a/examples/C/Makefile.am
+++ b/examples/C/Makefile.am
@@ -17,7 +17,7 @@ AM_LDFLAGS = ${top_builddir}/liblib/libnetcdf.la
 # These are the netCDF-3 examples.
 check_PROGRAMS = simple_xy_wr simple_xy_rd sfc_pres_temp_wr	\
 sfc_pres_temp_rd pres_temp_4D_wr pres_temp_4D_rd format		\
-quick_large_files
+quick_large_files quick_small_files
 
 TESTS = run_examples.sh format quick_large_files
 

--- a/examples/C/Makefile.am
+++ b/examples/C/Makefile.am
@@ -22,9 +22,8 @@ quick_large_files quick_small_files sfc_pres_temp_more simple
 TESTS = run_examples.sh format quick_large_files quick_small_files	\
 sfc_pres_temp_more simple
 
-# To build netcdf-4, or not to build netcdf-4, that is the question...
+# If HDF5 is being used, build the netcdf-4/HDF5 examples.
 if USE_HDF5
-# These are the netCDF-4 examples.
 check_PROGRAMS += simple_nc4_wr simple_nc4_rd simple_xy_nc4_wr	\
 simple_xy_nc4_rd
 

--- a/examples/C/Makefile.am
+++ b/examples/C/Makefile.am
@@ -17,10 +17,14 @@ AM_LDFLAGS = ${top_builddir}/liblib/libnetcdf.la
 # These are the netCDF-3 examples.
 check_PROGRAMS = simple_xy_wr simple_xy_rd sfc_pres_temp_wr	\
 sfc_pres_temp_rd pres_temp_4D_wr pres_temp_4D_rd format		\
-quick_large_files quick_small_files sfc_pres_temp_more simple
-
-TESTS = run_examples.sh format quick_large_files quick_small_files	\
 sfc_pres_temp_more simple
+
+TESTS = run_examples.sh format sfc_pres_temp_more simple
+
+if LARGE_FILE_TESTS
+check_PROGRAMS += quick_large_files quick_small_files
+TESTS += quick_large_files quick_small_files
+endif # LARGE_FILE_TESTS
 
 # If HDF5 is being used, build the netcdf-4/HDF5 examples.
 if USE_HDF5

--- a/examples/C/Makefile.am
+++ b/examples/C/Makefile.am
@@ -26,6 +26,7 @@ sfc_pres_temp_more simple
 if USE_HDF5
 check_PROGRAMS += simple_nc4_wr simple_nc4_rd simple_xy_nc4_wr	\
 simple_xy_nc4_rd
+TESTS += run_examples4.sh
 
 if ENABLE_FILTER_TESTING
 # filter_example.c should be same as nc_test4/test_filter.c
@@ -43,13 +44,10 @@ TESTS += run_par_test.sh
 endif #USE_PNETCDF
 endif # TEST_PARALLEL4
 
-if USE_HDF5
-TESTS += run_examples4.sh
-endif #USE_HDF5
-
 # These files are created by the tests.
 CLEANFILES = *.nc
 
-EXTRA_DIST = CMakeLists.txt run_examples.sh run_examples4.sh run_filter.sh
+EXTRA_DIST = CMakeLists.txt run_examples.sh run_examples4.sh	\
+run_filter.sh
 
 DISTCLEANFILES = findplugin.sh

--- a/examples/C/Makefile.am
+++ b/examples/C/Makefile.am
@@ -19,7 +19,8 @@ check_PROGRAMS = simple_xy_wr simple_xy_rd sfc_pres_temp_wr	\
 sfc_pres_temp_rd pres_temp_4D_wr pres_temp_4D_rd format		\
 quick_large_files quick_small_files sfc_pres_temp_more
 
-TESTS = run_examples.sh format quick_large_files quick_small_files
+TESTS = run_examples.sh format quick_large_files quick_small_files	\
+sfc_pres_temp_more
 
 # To build netcdf-4, or not to build netcdf-4, that is the question...
 if USE_HDF5

--- a/examples/C/Makefile.am
+++ b/examples/C/Makefile.am
@@ -17,9 +17,9 @@ AM_LDFLAGS = ${top_builddir}/liblib/libnetcdf.la
 # These are the netCDF-3 examples.
 check_PROGRAMS = simple_xy_wr simple_xy_rd sfc_pres_temp_wr	\
 sfc_pres_temp_rd pres_temp_4D_wr pres_temp_4D_rd format		\
-quick_large_files quick_small_files
+quick_large_files quick_small_files sfc_pres_temp_more
 
-TESTS = run_examples.sh format quick_large_files
+TESTS = run_examples.sh format quick_large_files quick_small_files
 
 # To build netcdf-4, or not to build netcdf-4, that is the question...
 if USE_HDF5

--- a/examples/C/Makefile.am
+++ b/examples/C/Makefile.am
@@ -20,7 +20,7 @@ sfc_pres_temp_rd pres_temp_4D_wr pres_temp_4D_rd format		\
 quick_large_files quick_small_files sfc_pres_temp_more simple
 
 TESTS = run_examples.sh format quick_large_files quick_small_files	\
-sfc_pres_temp_more
+sfc_pres_temp_more simple
 
 # To build netcdf-4, or not to build netcdf-4, that is the question...
 if USE_HDF5

--- a/examples/C/quick_large_files.c
+++ b/examples/C/quick_large_files.c
@@ -39,9 +39,9 @@ int
 main()
 {
     int ncid, spockid, kirkid, dimids[NUMDIMS];
-    double val_in, val_out = 999.99;
+    double val_out = 999.99;
     size_t index[NUMDIMS] = {DIM_LEN-1};
-    int i, res;
+    int res;
 
     /* Create the netCDF 64-bit offset format file. */
     if ((res = nc_create("example.nc", NC_CLOBBER|NC_64BIT_OFFSET, &ncid)))

--- a/examples/C/quick_small_files.c
+++ b/examples/C/quick_small_files.c
@@ -37,8 +37,9 @@ return NC_EXAMPLE_ERROR; \
 #define NUMVARS 2
 /* This dim len is the max size the first of two fixed size variables
    for an 8-byte type in classic format: int((2*31 - 4) / 8), that is,
-   2 GB minus a bit, on an 8 byte boundary. */
-#define DIM_LEN 268435455
+   2 GB minus a bit, on an 8 byte boundary. That number is
+   268435455. But this is the longest dim len which works. */
+#define DIM_LEN 268435440
 
 int
 main()

--- a/examples/C/quick_small_files.c
+++ b/examples/C/quick_small_files.c
@@ -44,9 +44,9 @@ int
 main()
 {
    int ncid, spockid, kirkid, dimids[NUMDIMS];
-   double val_in, val_out = 999.99;
+   double val_out = 999.99;
    size_t index[NUMDIMS] = {1};
-   int i, res;
+   int res;
 
    /* Create the netCDF classic format file. */
    if ((res = nc_create("example.nc", NC_CLOBBER, &ncid)))

--- a/examples/C/sfc_pres_temp_more.c
+++ b/examples/C/sfc_pres_temp_more.c
@@ -134,6 +134,8 @@ main()
         return retval;
     if (!(att_in = malloc(len_in)))
         return NC_ENOMEM;
+    if ((retval = nc_get_att_text(ncid, NC_GLOBAL, SONNET_NAME, att_in)))
+        return retval;
     if (strcmp(att_in, poem))
         error++;
     free(att_in);

--- a/examples/C/sfc_pres_temp_more.c
+++ b/examples/C/sfc_pres_temp_more.c
@@ -9,9 +9,11 @@
  * pressure and temperatures, and stores additional metadata as
  * dimension variables, an attribute.
  *
- * @author Ed Hartnett 2006/03/25
+ * @author Ed Hartnett started 2006/03/25, finished 2019/7/2
  */
 #include <netcdf.h>
+#include <stdlib.h>
+#include <string.h>
 
 #define FILE_NAME "sfc_pres_temp_more.nc"
 #define NDIMS 2
@@ -113,9 +115,9 @@ main()
         return retval;
 
     /* Write the phoney data. */
-    if ((retval = nc_put_var_float(ncid, pres_varid, pres_out)))
+    if ((retval = nc_put_var_float(ncid, pres_varid, (float *)pres_out)))
         return retval;
-    if ((retval = nc_put_var_float(ncid, temp_varid, temp_out)))
+    if ((retval = nc_put_var_float(ncid, temp_varid, (float *)temp_out)))
         return retval;
 
     /* Close the file. */
@@ -132,16 +134,16 @@ main()
         return retval;
     if (!(att_in = malloc(len_in)))
         return NC_ENOMEM;
-    if (strcmp(att_in, data_poem))
+    if (strcmp(att_in, poem))
         error++;
     free(att_in);
     if (error)
         return -2;
 
     /* Read the data. */
-    if ((retval = nc_get_var_float(ncid, pres_varid, pres_in)))
+    if ((retval = nc_get_var_float(ncid, pres_varid, (float *)pres_in)))
         return retval;
-    if ((retval = nc_get_var_float(ncid, temp_varid, temp_in)))
+    if ((retval = nc_get_var_float(ncid, temp_varid, (float *)temp_in)))
         return retval;
 
     /* Check the data. */

--- a/examples/C/sfc_pres_temp_more.c
+++ b/examples/C/sfc_pres_temp_more.c
@@ -10,10 +10,10 @@
  * dimension variables, an attribute.
  *
  * @author Ed Hartnett 2006/03/25
-*/
+ */
 #include <netcdf.h>
 
-#define FILE_NAME basic1.nc
+#define FILE_NAME "sfc_pres_temp_more.nc"
 #define NDIMS 2
 #define LAT_NAME "latitude"
 #define LAT_LEN 40
@@ -23,136 +23,137 @@
 #define TEMP_NAME "surface_temperature"
 #define SONNET_NAME "Brownings_sonnet_mangled_by_Hartnett"
 
-char data_poem[] = "How do I love data? Let me count the ways.\n
-I love data to the depth and breadth and height\n
-My model can reach, when running overnight\n
-From the ends of CONUS and ideal Space.\n
-I love data to the level of every day's\n
-6Z ob, by Sun and Linux box.\n
-I love data freely, as PIs strive for funding;\n
-I love data purely, as good C code is compiled,\n
-I love data with the passion put to use\n
-In my old programs, and with my childhood's TRS-80.\n
-I love data with a love I seemed to lose\n
-With my lost password -I love data with the constants,\n
-expressions, statements, of all my code! -and, if NSF choose,\n
-I shall love data better after processing.";
+char poem[] = "How do I love data? Let me count the ways.\n"
+    "I love data to the depth and breadth and height\n"
+    "My model can reach, when running overnight\n"
+    "From the ends of CONUS and ideal Space.\n"
+    "I love data to the level of every day's\n"
+    "6Z ob, by Sun and Linux box.\n"
+    "I love data freely, as PIs strive for funding;\n"
+    "I love data purely, as good C code is compiled,\n"
+    "I love data with the passion put to use\n"
+    "In my old programs, and with my childhood's TRS-80.\n"
+    "I love data with a love I seemed to lose\n"
+    "With my lost password -I love data with the constants,\n"
+    "expressions, statements, of all my code! -and, if NSF choose,\n"
+    "I shall love data better after processing.";
 
 int
 main()
 {
-   int ncid, lon_dimid, lat_dimid, pres_varid, temp_varid;
-   int dimids[NDIMS];
-   float pres_out[LAT_LEN][LON_LEN], pres_in[LAT_LEN][LON_LEN];
-   float temp_out[LAT_LEN][LON_LEN], temp_in[LAT_LEN][LON_LEN];
-   char *att_in;
-   size_t len_in;
-   int error = 0;
-   int lat, lon, retval;
+    int ncid, lon_dimid, lat_dimid, pres_varid, temp_varid, lat_varid, lon_varid;
+    int dimids[NDIMS];
+    float pres_out[LAT_LEN][LON_LEN], pres_in[LAT_LEN][LON_LEN];
+    float temp_out[LAT_LEN][LON_LEN], temp_in[LAT_LEN][LON_LEN];
+    float latitude[LAT_LEN], longitude[LON_LEN];
+    char *att_in;
+    size_t len_in;
+    int error = 0;
+    int lat, lon, retval;
 
-   /* Create phoney data. If this wasn't an example program, we would
-    * have some real data to write, for example, model output. */
-   for (lat = 0; lat < LAT_LEN; lat++)
-      for (lon = 0; lon < LON_LEN; lon++)
-      {
-	 pres_out[lat][lon] = 1013.1;
-	 temp_out[lat][lon] = 12.5;
-      }
+    /* Create phoney data. If this wasn't an example program, we would
+     * have some real data to write, for example, model output. */
+    for (lat = 0; lat < LAT_LEN; lat++)
+        for (lon = 0; lon < LON_LEN; lon++)
+        {
+            pres_out[lat][lon] = 1013.1;
+            temp_out[lat][lon] = 12.5;
+        }
 
-   /* These are the latitudes and longitudes which correspond with
-    * ticks on the dimension axes. */
-   for (lat = 0; lat < LAT_LEN; lat++)
-      latitude[lat] = 40. + lat * 2.5;
-   for (lon = 0; lon < LON_LEN; lon++)
-      longitude[lon] = -90. - lon * 5;
+    /* These are the latitudes and longitudes which correspond with
+     * ticks on the dimension axes. */
+    for (lat = 0; lat < LAT_LEN; lat++)
+        latitude[lat] = 40. + lat * 2.5;
+    for (lon = 0; lon < LON_LEN; lon++)
+        longitude[lon] = -90. - lon * 5;
 
-   /* Create the file. */
-   if ((retval = nc_create(FILE_NAME, NC_CLOBBER, &ncid)))
-      return retval;
+    /* Create the file. */
+    if ((retval = nc_create(FILE_NAME, NC_CLOBBER, &ncid)))
+        return retval;
 
-   /* Add data sonnet. By using NC_GLOBAL we mean that this attribute
-    * applies to the entire file, not just to one variable. Don't
-    * forget that sizeof does not include the null terminator, so if
-    * you want it, you need to add one more byte. */
-   if ((retval = nc_put_att_text(ncid, NC_GLOBAL, SONNET_NAME,
-				 sizeof(poem) + 1, poem)))
-      return retval;
+    /* Add data sonnet. By using NC_GLOBAL we mean that this attribute
+     * applies to the entire file, not just to one variable. Don't
+     * forget that sizeof does not include the null terminator, so if
+     * you want it, you need to add one more byte. */
+    if ((retval = nc_put_att_text(ncid, NC_GLOBAL, SONNET_NAME,
+                                  sizeof(poem) + 1, poem)))
+        return retval;
 
-   /* Define the dimensions. */
-   if ((retval = nc_def_dim(ncid, LAT_NAME, LAT_LEN, &lat_dimid)))
-      return retval;
-   if ((retval = nc_def_dim(ncid, LON_NAME, LON_LEN, &lon_dimid)))
-      return retval;
+    /* Define the dimensions. */
+    if ((retval = nc_def_dim(ncid, LAT_NAME, LAT_LEN, &lat_dimid)))
+        return retval;
+    if ((retval = nc_def_dim(ncid, LON_NAME, LON_LEN, &lon_dimid)))
+        return retval;
 
-   /* Save the dimension information, in variables of the same
-    * name. First we need to define these variables. */
-   dimids[0] = lat_dimid;
-   if ((retval = nc_def_var(ncid, LAT_NAME, NC_FLOAT, 1, dimids, &lat_varid)))
-      return retval;
-   dimids[0] = lon_dimid;
-   if ((retval = nc_def_var(ncid, LON_NAME, NC_FLOAT, 1, dimids, &lon_varid)))
-      return retval;
+    /* Save the dimension information, in variables of the same
+     * name. First we need to define these variables. */
+    dimids[0] = lat_dimid;
+    if ((retval = nc_def_var(ncid, LAT_NAME, NC_FLOAT, 1, dimids, &lat_varid)))
+        return retval;
+    dimids[0] = lon_dimid;
+    if ((retval = nc_def_var(ncid, LON_NAME, NC_FLOAT, 1, dimids, &lon_varid)))
+        return retval;
 
-   /* Define the variables. */
-   dimids[0] = lat_dimid;
-   dimids[1] = lon_dimid;
-   if ((retval = nc_def_var(ncid, PRES_NAME, NC_FLOAT, NDIMS, dimids, &pres_varid)))
-      return retval;
-   if ((retval = nc_def_var(ncid, TEMP_NAME, NC_FLOAT, NDIMS, dimids, &temp_varid)))
-      return retval;
+    /* Define the variables. */
+    dimids[0] = lat_dimid;
+    dimids[1] = lon_dimid;
+    if ((retval = nc_def_var(ncid, PRES_NAME, NC_FLOAT, NDIMS, dimids, &pres_varid)))
+        return retval;
+    if ((retval = nc_def_var(ncid, TEMP_NAME, NC_FLOAT, NDIMS, dimids, &temp_varid)))
+        return retval;
 
-   /* End define mode. */
-   if ((retval = nc_enddef(ncid)))
-      return retval;
+    /* End define mode. */
+    if ((retval = nc_enddef(ncid)))
+        return retval;
 
-   /* Write the dimension metadata. */
-   if ((retval = nc_put_var_float(ncid, lat_varid, latitude_out)))
-      return retval;
-   if ((retval = nc_put_var_float(ncid, lon_varid, longitude_out)))
-      return retval;
+    /* Write the dimension metadata. */
+    if ((retval = nc_put_var_float(ncid, lat_varid, latitude)))
+        return retval;
+    if ((retval = nc_put_var_float(ncid, lon_varid, longitude)))
+        return retval;
 
-   /* Write the phoney data. */
-   if ((retval = nc_put_var_float(ncid, pres_varid, pres_out)))
-      return retval;
-   if ((retval = nc_put_var_float(ncid, temp_varid, temp_out)))
-      return retval;
+    /* Write the phoney data. */
+    if ((retval = nc_put_var_float(ncid, pres_varid, pres_out)))
+        return retval;
+    if ((retval = nc_put_var_float(ncid, temp_varid, temp_out)))
+        return retval;
 
-   /* Close the file. */
-   if ((retval = nc_close(ncid)))
-      return retval;
+    /* Close the file. */
+    if ((retval = nc_close(ncid)))
+        return retval;
 
-   /* Open the file and check that everything's OK. */
-   if ((retval = nc_open(FILE_NAME, 0, &ncid)))
-      return retval;
+    /* Open the file and check that everything's OK. */
+    if ((retval = nc_open(FILE_NAME, 0, &ncid)))
+        return retval;
 
-   /* Read the attribute. First find it's length to allocate storage
-    * for it. */
-   if ((retval = nc_inq_attlen(ncid, NC_GLOBAL, SONNET_NAME, &len_in)))
-      return retval;
-   if (!(att_in = malloc(len_in)))
-      return NC_ENOMEM;
-   if (strcmp(att_in, data_poem))
-      error++;
-   free(att_in);
-   if (error)
-      return -2;
+    /* Read the attribute. First find it's length to allocate storage
+     * for it. */
+    if ((retval = nc_inq_attlen(ncid, NC_GLOBAL, SONNET_NAME, &len_in)))
+        return retval;
+    if (!(att_in = malloc(len_in)))
+        return NC_ENOMEM;
+    if (strcmp(att_in, data_poem))
+        error++;
+    free(att_in);
+    if (error)
+        return -2;
 
-   /* Read the data. */
-   if ((retval = nc_get_var_float(ncid, pres_varid, pres_in)))
-      return retval;
-   if ((retval = nc_get_var_float(ncid, temp_varid, temp_in)))
-      return retval;
+    /* Read the data. */
+    if ((retval = nc_get_var_float(ncid, pres_varid, pres_in)))
+        return retval;
+    if ((retval = nc_get_var_float(ncid, temp_varid, temp_in)))
+        return retval;
 
-   /* Check the data. */
-   for (lat = 0; lat < LAT_LEN; lat++)
-      for (lon = 0; lon < LON_LEN; lon++)
-	 if (pres_in[lat][lon] != pres_out[lat][lon] ||
-	     temp_in[lat][lon] != temp_out[lat][lon])
-	    return -2;
+    /* Check the data. */
+    for (lat = 0; lat < LAT_LEN; lat++)
+        for (lon = 0; lon < LON_LEN; lon++)
+            if (pres_in[lat][lon] != pres_out[lat][lon] ||
+                temp_in[lat][lon] != temp_out[lat][lon])
+                return -2;
 
-   /* Close the file. */
-   if ((retval = nc_close(ncid)))
-      return retval;
+    /* Close the file. */
+    if ((retval = nc_close(ncid)))
+        return retval;
 
-   return 0;
+    return 0;
 }

--- a/examples/C/simple.c
+++ b/examples/C/simple.c
@@ -1,12 +1,10 @@
 /*! \file
 
-Copyright 1993, 1994, 1995, 1996, 1997, 1998, 1999, 2000, 2001, 2002,
-2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014,
-2015, 2016, 2017, 2018
-University Corporation for Atmospheric Research/Unidata.
-
-See \ref copyright file for more info.
-
+  Copyright 1993, 1994, 1995, 1996, 1997, 1998, 1999, 2000, 2001, 2002,
+  2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014,
+  2015, 2016, 2017, 2018
+  University Corporation for Atmospheric Research/Unidata.
+  See \ref copyright file for more info.
 */
 
 /* This example program is part of Unidata's netCDF library for
@@ -15,7 +13,6 @@ See \ref copyright file for more info.
    How about a short, but meaningful, netCDF program?
 
    Ed Hartnett, 6/19/4
-   $Id: simple.c,v 1.1 2004/07/26 14:04:42 ed Exp $
 */
 
 #include <netcdf.h>
@@ -25,11 +22,11 @@ See \ref copyright file for more info.
 /* This macro handles errors by outputting a message to stdout and
    then exiting. */
 #define NC_EXAMPLE_ERROR 2 /* This is the exit code for failure. */
-#define BAIL(e) do { \
-printf("Bailing out in file %s, line %d, error:%s.\n", \
-__FILE__, __LINE__, nc_strerror(e)); \
-return NC_EXAMPLE_ERROR; \
-} while (0)
+#define BAIL(e) do {                                            \
+        printf("Bailing out in file %s, line %d, error:%s.\n",  \
+               __FILE__, __LINE__, nc_strerror(e));             \
+        return NC_EXAMPLE_ERROR;                                \
+    } while (0)
 
 #define NUMDIMS 2
 #define NUMVARS 1
@@ -40,45 +37,45 @@ return NC_EXAMPLE_ERROR; \
 int
 main()
 {
-   int ncid, temp_varid, dimids[NUMDIMS];
-   float temp[LAT_LEN][LON_LEN], *fp;
-   int i, res;
+    int ncid, temp_varid, dimids[NUMDIMS];
+    float temp[LAT_LEN][LON_LEN], *fp;
+    int i, res;
 
-   /* Create a bunch of phoney data so we have something to write in
-      the example file. */
-   for (fp=(float *)temp, i=0; i<LAT_LEN*LON_LEN; i++)
-      *fp++ = 10. + i/10.;
+    /* Create a bunch of phoney data so we have something to write in
+       the example file. */
+    for (fp=(float *)temp, i=0; i<LAT_LEN*LON_LEN; i++)
+        *fp++ = 10. + i/10.;
 
-   /* Create the netCDF file. */
-   if ((res = nc_create("short.nc", NC_CLOBBER, &ncid)))
-      BAIL(res);
+    /* Create the netCDF file. */
+    if ((res = nc_create("short.nc", NC_CLOBBER, &ncid)))
+        BAIL(res);
 
-   /* Define dimensions. */
-   if ((res = nc_def_dim(ncid, "latitude", LAT_LEN, dimids)))
-      BAIL(res);
-   if ((res = nc_def_dim(ncid, "longitude", LON_LEN, &dimids[1])))
-      BAIL(res);
+    /* Define dimensions. */
+    if ((res = nc_def_dim(ncid, "latitude", LAT_LEN, dimids)))
+        BAIL(res);
+    if ((res = nc_def_dim(ncid, "longitude", LON_LEN, &dimids[1])))
+        BAIL(res);
 
-   /* Define the variable. */
-   if ((res = nc_def_var(ncid, "sfc_temp", NC_FLOAT, NUMDIMS,
-			 dimids, &temp_varid)))
-      BAIL(res);
+    /* Define the variable. */
+    if ((res = nc_def_var(ncid, "sfc_temp", NC_FLOAT, NUMDIMS,
+                          dimids, &temp_varid)))
+        BAIL(res);
 
-   /* We'll store the units. */
-   if ((res = nc_put_att_text(ncid, temp_varid, "units",
-			      strlen(CELSIUS), CELSIUS)))
-      BAIL(res);
+    /* We'll store the units. */
+    if ((res = nc_put_att_text(ncid, temp_varid, "units",
+                               strlen(CELSIUS), CELSIUS)))
+        BAIL(res);
 
-   /* We're finished defining metadata. */
-   if ((res = nc_enddef(ncid)))
-      BAIL(res);
+    /* We're finished defining metadata. */
+    if ((res = nc_enddef(ncid)))
+        BAIL(res);
 
-   if ((res = nc_put_var_float(ncid, temp_varid, (float *)temp)))
-      BAIL(res);
+    if ((res = nc_put_var_float(ncid, temp_varid, (float *)temp)))
+        BAIL(res);
 
-   /* We're done! */
-   if ((res = nc_close(ncid)))
-      BAIL(res);
+    /* We're done! */
+    if ((res = nc_close(ncid)))
+        BAIL(res);
 
-   return 0;
+    return 0;
 }

